microcosm_pubsub/chain/chain.py

```diff
@@ -0,0 +1,65 @@
+from marshmallow import ValidationError
+
+from microcosm_pubsub.chain.context_decorators import (
+    get_from_context,
+    save_to_context,
+    save_to_context_by_func_name,
+    temporary_replace_context_keys,
+)
+
+
+class Chain:
+    """
+    Structure that contains callable functions that shares the same context.
+
+    """
+    def __init__(self, *args):
+        """
+        Build a chain that can be resolved multiple times
+        :param *args: callable functions, Chain statemnts or other Chains
+
+        """
+        self.pieces = args
+
+        self.context_decorators = [
```

Comment:
We can easily extended the chain by passing more decorators here. even inheritance should work.

Reply:
I don't follow: do you expect subclasses to `append/extend` this structure? I think we can design a better API, e.g. as an `@property`